libcrux-ml-dsa/src/simd/avx2.rs

```diff
@@ -7,163 +10,151 @@ mod ntt;
 mod rejection_sample;
 mod vector_type;
 
-pub(crate) use vector_type::AVX2SIMDUnit;
+pub(crate) use vector_type::{AVX2RingElement, AVX2SIMDUnit};
 
+/// Implementing the [`Operations`] for AVX2.
 impl Operations for AVX2SIMDUnit {
+    type Coefficient = vector_type::Vec256;
+
     #[inline(always)]
-    fn ZERO() -> Self {
-        vector_type::ZERO()
+    fn zero() -> Self::Coefficient {
+        vector_type::zero()
     }
 
     #[inline(always)]
-    fn from_coefficient_array(coefficient_array: &[i32]) -> Self {
-        vector_type::from_coefficient_array(coefficient_array)
+    fn from_coefficient_array(coefficient_array: &[i32], out: &mut Self::Coefficient) {
+        vector_type::from_coefficient_array(coefficient_array, out)
     }
 
     #[inline(always)]
-    fn to_coefficient_array(&self) -> [i32; 8] {
-        vector_type::to_coefficient_array(&self)
+    fn to_coefficient_array(value: &Self::Coefficient, out: &mut [i32]) {
+        vector_type::to_coefficient_array(value, out)
     }
+
     #[inline(always)]
-    fn add(lhs: &Self, rhs: &Self) -> Self {
-        arithmetic::add(lhs.coefficients, rhs.coefficients).into()
+    fn add(lhs: &mut Self::Coefficient, rhs: &Self::Coefficient) {
```

Comment:
I guess the switch to using a borrow for the RHS and a mutable borrow for the LHS is primarily to have consistency with the portable version so we can share the Operations trait between them? Might be worth a comment.
Otherwise it looks a bit weird for AVX2 since Vec256 is a scalar.